src/core/qgssnappingconfig.h

```diff
@@ -335,6 +335,12 @@ class CORE_EXPORT QgsSnappingConfig
     //! Sets if the snapping on intersection is enabled
     void setIntersectionSnapping( bool enabled );
 
+    //! Returns if self snapping is enabled
```

Comment:
Can you add a minimal description of what "self snapping" is and add a `\since QGIS 3.14`